astropy/units/astrophys.py

```diff
@@ -70,15 +70,16 @@
          doc="Electron mass",
          format={'latex': r'M_{e}', 'unicode': 'Mₑ'})
 # Unified atomic mass unit
-def_unit(['u', 'Da', 'Dalton'], 1.6605387e-27 * si.kg, namespace=_ns,
+def_unit(['u', 'Da', 'Dalton'], _si.u.value * si.kg, namespace=_ns,
          prefixes=True, exclude_prefixes=['a', 'da'],
          doc="Unified atomic mass unit")
 
 ##########################################################################
 # ENERGY
 
-def_unit(['Ry', 'rydberg'], 13.605692 * si.eV, namespace=_ns,
-         prefixes=True,
+def_unit(['Ry', 'rydberg'],
+         _si.Ryd.value * _si.c.value * _si.h.value / _si.e.value * si.eV,
+         namespace=_ns, prefixes=True,
```

Comment:
One other thing we might want to consider, not  for this PR, but in general, is maybe allowing constants to be defined in different units--override the `.to` method to give hard-coded values for the constant in some given units rather than performing the (sometimes inaccurate) conversion.

For example the main reason for the discrepancies above is in our conversion of `h` (which we have defined in `J s`) to `eV s`, which ends up off a bit from the value given in CODATA2010 (albeit within the uncertainty, which wouldn't matter except when we then go and use it to derive other constants).